service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java

```diff
@@ -0,0 +1,27 @@
+package org.oskari.service.wfs.client;
+
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.oskari.service.wfs3.client.OskariWFS3Client;
+
+public class OskariWFSClient {
+
+    private static final String WFS_3_VERSION = "3.0.0";
+
+    private OskariWFSClient() {}
+
+    public static SimpleFeatureCollection tryGetFeatures(
```

Comment:
Any reason to have try at the start of the method name instead of just getFeatures? 

I'd really like this to work like OskariWFSClient.getVersion(ver).getFeatures(endPoint, user, pass, typeName, bbox, crs, maxFeatures) or maybe even something like OskariWFSClient.create(endPoint, user, pass, ver).useCache(cache).getFeatures(typeName, bbox, crs, maxFeatures), but this would mean making a lot of static methods non-static and I'm not sure if it's worth it. 

Maybe it would enable better control for the amount of requests sent to a specific endpoint and we might want to wrap it as a Hystrix command etc. But... maybe we can use this as is and worry about DOSing services in the next version :)

Reply:
The try-prefix is trying to denote that the function is trying to do something, and that it might not succeed in doing it, but it will not throw an Exception if it fails, but rather it handles the possible errors itself and returns null on failure (could also be Optional<T>).

This class is nothing but a dumb proxy, we could make this class more "client-y" pretty easily. This class could handle all those caching, request pooling etc. and we wouldn't have to touch OskariWFS110Client and OskariWFS3Client at all. Those could still be just a set of static functions. The caching and request pooling mechanisms should probably work the same way independent of the version of the underlying WFS services.